model/API.inc.php

```diff
@@ -1323,71 +1323,47 @@ public static function getSearchParamValues($params, $param) {
 		
 		return $sets;
 	}
-
-	public static function getRequestLimits($params) {
-		$rate = null;
-		$concurrent = null;
-
-		if(Z_CONFIG::$REDUCE_LOAD) {
-			if ($params['userID']) {
-				$rate = [
-					'logOnly' => false,
-					'bucket' => $params['userID'] . '_' . $params['ip'],
-					'limit' => 1, //request per second
-					'burst' => 3, //burst multiplier for 'limit'
-					'warn' => 1 //remaining requests threshold to warn client
-				];
-
-				$concurrent = [
-					'logOnly' => false,
-					'bucket' => $params['userID'],
-					'ttl' => 10,
-					'limit' => 2,
-					'warn' => 1
-				];
-			} else {
-				$rate = [
-					'logOnly' => false,
-					'bucket' => $params['ip'],
-					'limit' => 10,
-					'burst' => 1,
-					'warn' => 50
-				];
-			}
-		} else {
-			if ($params['userID']) {
-				$rate = [
-					'logOnly' => false,
-					'bucket' => $params['userID'] . '_' . $params['ip'],
-					'limit' => 1,
-					'burst' => 3,
-					'warn' => 1
-				];
-
-				$concurrent = [
-					'logOnly' => false,
-					'bucket' => $params['userID'],
-					'ttl' => 10,
-					'limit' => 2,
-					'warn' => 1
-				];
-			} else {
-				$rate = [
-					'logOnly' => false,
-					'bucket' => $params['ip'],
-					'limit' => 100,
-					'burst' => 3,
-					'warn' => 50
-				];
-			}
+	
+	public static function getRateLimit($userID, $ip) {
```

Comment:
Well, I thought it's cleaner because there isn't going to be too many parameters and you can clearly see what that function uses to calculate the limit.